lms/services/canvas_api/client.py

```diff
@@ -205,18 +208,40 @@ def list_files(self, course_id):
         Return the list of files for the given `course_id`.
 
         :param course_id: the Canvas course_id of the course to look in
+        :param request: Pyramid request object used to store file details if
+            present
         :rtype: list(dict)
         """
         # For documentation of this request see:
         # https://canvas.instructure.com/doc/api/files.html#method.files.api_index
 
-        return self._client.send(
+        files = self._client.send(
             "GET",
             f"courses/{course_id}/files",
             params={"content_types[]": "application/pdf"},
             schema=self._ListFilesSchema,
         )
 
+        if self._request:
```

Comment:
Should it take a `store` or similar now? `_request` will always be here.

Reply:
I think maybe `request` just shouldn't be optional.